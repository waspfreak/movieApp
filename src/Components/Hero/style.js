import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	overflow-y: hidden;
	overflow-x: scroll;
`;

export const HeroStyles = styled.div`
	background-color: #111;
`;

export const HeroItemStyles = styled.div`
	transition: transform 0.45s;
	:hover {
		transform: scale(1.09);
	}
`;

export const HeroTitle = styled.h2`
	font-family: 'Titillium Web', sans-serif;
	color: #fff;
	display: inline-block;
	font-size: 1.8em;
	padding-left: 20px;
	padding-top: 16px;
	padding-bottom: 26px;
	font-weight: bold;
`;
