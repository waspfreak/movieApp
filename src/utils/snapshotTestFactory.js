import {cleanup, render} from '@testing-library/react';
import React from 'react';

/**
 *
 * Sets up snapshot tests in a standard way and avoids repeating boiler plate
 */
export function snapshotTestFactory(Component: any, properties: any) {
	/**
	 * Re-usable it block
	 */
	const itBlock = (props: any) => {
		it('should match snapshot', () => {
			const {container} = render(<Component {...props} />);
			expect(container).toMatchSnapshot();
		});
	};

	/**
	 * Re-usable describe block
	 */
	describe(`${Component.name}`, () => {
		/**
		 * Clean up any mock functions that might be in props
		 */
		afterEach(() => {
			jest.clearAllMocks();
			cleanup();
		});

		/**
		 * If its an array of props then iterate them into seperate it blocks
		 */
		if (Array.isArray(properties)) {
			properties.forEach((props) => {
				itBlock(props);
			});
		} else {
			itBlock(properties);
		}
	});
}
