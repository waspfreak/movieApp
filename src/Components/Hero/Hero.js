import React, {useState} from 'react';
import SimpleCard from '../Card/SimpleCard';
import {HeroStyles, HeroItemStyles, HeroTitle, Container} from './style';
//import {useNowPlayingFetch} from '../../hooks/useNowPlayingFetch';
import {useHomeFetch} from '../../hooks/useHomeFetch';

// Environment Variables
import {IMAGE_BASE_URL, POSTER_SIZE} from '../../api/API';

// Images
import NoImage from '../../assets/no_image.jpg';

export const Hero = (posterPath) => {
	const [searchTerm, setSearchTerm] = useState('');
	const [
		{
			state: {movies},
		},
	] = useHomeFetch(searchTerm);

	return (
		<HeroStyles>
			<HeroTitle>Popular Movies</HeroTitle>
			<Container>
				{movies.map((movie) => (
					<>
						<HeroItemStyles>
							<SimpleCard
								posterPath={
									movie.poster_path
										? `${IMAGE_BASE_URL}${POSTER_SIZE}${movie.poster_path}`
										: NoImage
								}
							/>
						</HeroItemStyles>
					</>
				))}
			</Container>
		</HeroStyles>
	);
};

export default Hero;
