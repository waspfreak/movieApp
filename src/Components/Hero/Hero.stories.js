import React from 'react';

import {Hero} from './Hero';

export default {
	title: 'Example/Hero',
	component: Hero,
};

const Template = (args) => <Hero {...args} />;

export const HeroExample = Template.bind({});
HeroExample.args = {};
