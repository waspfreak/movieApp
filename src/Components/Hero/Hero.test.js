import {snapshotTestFactory} from '../../utils/snapshotTestFactory';
//
import {Hero} from './Hero';
const props: Hero[] = [{}];
snapshotTestFactory(Hero, props);
