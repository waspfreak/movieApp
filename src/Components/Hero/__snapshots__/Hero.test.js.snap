// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should match snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-y: hidden;
  overflow-x: scroll;
}

.c0 {
  background-color: #111;
}

.c1 {
  font-family: 'Titillium Web',sans-serif;
  color: #fff;
  display: inline-block;
  font-size: 1.8em;
  padding-left: 20px;
  padding-top: 16px;
  padding-bottom: 26px;
  font-weight: bold;
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      Popular Movies
    </h2>
    <div
      class="c2"
    >
      loading...
    </div>
  </div>
</div>
`;
